types/bootstrap/js/dist/offcanvas.d.ts

```diff
@@ -1,41 +1,87 @@
 import BaseComponent, { GetInstanceFactory, GetOrCreateInstanceFactory } from './base-component';
 
 declare class Offcanvas extends BaseComponent {
+    /**
+     * Static method which allows you to get the offcanvas instance associated with a DOM element
+     */
     static getInstance: GetInstanceFactory<Offcanvas>;
 
     /**
      * Static method which allows you to get the offcanvas instance associated with
      *  a DOM element, or create a new one in case it wasn’t initialised
      */
-    static getOrCreateInstance: GetOrCreateInstanceFactory<Offcanvas>;
+    static getOrCreateInstance: GetOrCreateInstanceFactory<Offcanvas, Partial<Offcanvas.Options>>;
 
     static jQueryInterface: Offcanvas.jQueryInterface;
 
+    constructor(element: string | Element, options?: Partial<Offcanvas.Options>);
+
+    /**
+     * Toggles an offcanvas element to shown or hidden. Returns to the caller before the offcanvas element has actually
+     * been shown or hidden (i.e. before the shown.bs.offcanvas or hidden.bs.offcanvas event occurs).
+     */
     toggle(relatedTarget?: HTMLElement): void;
 
+    /**
+     * Shows an offcanvas element. Returns to the caller before the offcanvas element has actually been shown
+     * (i.e. before the shown.bs.offcanvas event occurs).
+     */
     show(relatedTarget?: HTMLElement): void;
 
+    /**
+     * Hides an offcanvas element. Returns to the caller before the offcanvas element has actually been hidden
+     * (i.e. before the hidden.bs.offcanvas event occurs).
+     */
     hide(): void;
 }
 
 declare namespace Offcanvas {
-    type jQueryInterface = (config?: 'toggle' | 'show' | 'hide' | 'dispose') => void;
+    interface Options {
+        /**
+         * Apply a backdrop on body while offcanvas is open
+         *
+         * @default true
+         */
+        backdrop: boolean;
+
+        /**
+         * Closes the offcanvas when escape key is pressed
+         *
+         * @default true
+         */
+        keyboard: boolean;
+
+        /**
+         * 	Allow body scrolling while offcanvas is open
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/bootstrap/js/dist/offcanvas.d.ts;
  Line: 55;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.